tests/component/test_timing_properties.py

```diff
@@ -0,0 +1,221 @@
+"Contains a collection of pytest tests that validates the timing properties."
+
+import pytest
+
+import nidaqmx
+import nidaqmx.system
+from nidaqmx import DaqError
+from nidaqmx.constants import AcquisitionType, SampleTimingType
+from nidaqmx.error_codes import DAQmxErrors
+
+
+def test__timing__get_boolean_property__returns_value(any_x_series_device):
+    """Test for validating getter for boolean property."""
+    with nidaqmx.Task() as task:
+        task.ao_channels.add_ao_voltage_chan(any_x_series_device.ao_physical_chans[0].name)
+        task.timing.samp_timing_type = SampleTimingType.ON_DEMAND
+
+        assert not task.timing.simultaneous_ao_enable
+
+
+def test__timing__set_boolean_property__returns_assigned_value(any_x_series_device):
+    """Test for validating setter for boolean property."""
+    with nidaqmx.Task() as task:
+        task.ao_channels.add_ao_voltage_chan(any_x_series_device.ao_physical_chans[0].name)
+        task.timing.samp_timing_type = SampleTimingType.ON_DEMAND
+
+        task.timing.simultaneous_ao_enable = True
+
+        assert task.timing.simultaneous_ao_enable
+
+
+def test__timing__reset_boolean_property__returns_default_value(any_x_series_device):
+    """Test for validating reset for boolean property."""
+    with nidaqmx.Task() as task:
+        task.ao_channels.add_ao_voltage_chan(any_x_series_device.ao_physical_chans[0].name)
+        task.timing.samp_timing_type = SampleTimingType.ON_DEMAND
+
+        del task.timing.simultaneous_ao_enable
+
+        assert not task.timing.simultaneous_ao_enable
+
+
+def test__timing__getter_string_property__returns_value(any_x_series_device):
+    """Test for validating getter string property."""
+    with nidaqmx.Task() as task:
+        task.ai_channels.add_ai_voltage_chan(any_x_series_device.ai_physical_chans[0].name)
+        task.timing.cfg_samp_clk_timing(1000)
+
+        assert task.timing.samp_clk_src == f"/{any_x_series_device.name}/ai/SampleClockTimebase"
+
+
+def test__timing__setter_string_property__returns_assigned_value(any_x_series_device):
+    """Test for validating setter string property."""
+    with nidaqmx.Task() as task:
+        task.ai_channels.add_ai_voltage_chan(any_x_series_device.ai_physical_chans[0].name)
+        task.timing.cfg_samp_clk_timing(1000)
+
+        task.timing.samp_clk_src = "PFI0"
+
+        assert task.timing.samp_clk_src == f"/{any_x_series_device.name}/PFI0"
+
+
+def test__timing__reset_string_property__returns_default_value(any_x_series_device):
+    """Test for validating reset string property."""
+    with nidaqmx.Task() as task:
+        task.ai_channels.add_ai_voltage_chan(any_x_series_device.ai_physical_chans[0].name)
+        task.timing.cfg_samp_clk_timing(1000)
+
+        del task.timing.samp_clk_src
+
+        assert task.timing.samp_clk_src == f"/{any_x_series_device.name}/ai/SampleClockTimebase"
```

Comment:
Pass `cfg_samp_clk_timing(1000, source="PFI0")` in order to set `samp_clk_src` to a non-default value. If you set it the same as the default, reset has no effect.

Reply:
Updated the source name `PFI0` and reset and verified the default source name.